### PR TITLE
boards: imx_rt: Don't default FLASH_MCUX_FLEXSPI_XIP to enabled

### PR DESCRIPTION
**Pull request by danieldegrasse**
Do not default FLASH_MCUX_FLEXSPI_XIP to enabled when code is not located in flash,
this will cause issues if code is executing from ITCM, as the
zephyr_code_relocate macro will relocate the flash driver code into
itcm, and overwrite the zephyr image.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

### CODE REVIEW

File: boards/arm/mimxrt1020_evk/Kconfig.defconfig [hunk 1]
```diff
@@ -15,13 +15,17 @@ endchoice
...
 
+if FLASH
+
 config FLASH_MCUX_FLEXSPI_NOR
-	default y if FLASH && BOARD_MIMXRT1020_EVK
+	default y
 
 config FLASH_MCUX_FLEXSPI_XIP
-	default y if FLASH && BOARD_MIMXRT1020_EVK
+	default y if (CODE_FLEXSPI || CODE_FLEXSPI2)
```
Comment (mmahadevan108) [inline boards/arm/mimxrt1020_evk/Kconfig.defconfig:24]: `FLASH_MCUX_FLEXSPI_XIP` Kconfig already has this line defined,
`depends on (CODE_FLEXSPI || CODE_FLEXSPI2 || SOC_SERIES_IMX_RT6XX)`
Do we need ` if (CODE_FLEXSPI || CODE_FLEXSPI2)` here

**Comment by mmahadevan108** [comment]
Am I right in my understanding that this commit fixes an issue on the `RT11760` and `RT1170`. All others is code cleanup. If so, please update the commit message to reflect the platforms which had the issue.

**Comment by danieldegrasse** [comment]
> Am I right in my understanding that this commit fixes an issue on the RT1160 and RT1170. All others is code cleanup. If so, please update the commit message to reflect the platforms which had the issue.

@mmahadevan108 The issue was originally found by @DerekSnell on the RT1060, but it affects the majority of the RT 4 digit platforms. If code is running from RAM with `CONFIG_CODE_ITCM` or `CONFIG_CODE_SEMC`, then `CONFIG_FLASH_MCUX_FLEXSPI_XIP` should not be selected. 

Generally, `CONFIG_FLASH_MCUX_FLEXSPI` is not required if code is executing from RAM, but in the specific case of ITCM an issue showed up because the `zephyr_code_relocate` directive `CONFIG_FLASH_MCUX_FLEXSPI` enables was copying the code for the memc driver into ITCM, which overwrote the application image already stored there.

This PR updates the board defconfig files for all RT 4 digit platforms to ensure that `CONFIG_FLASH_MCUX_FLEXSPI` will only be enabled if `CONFIG_CODE_FLEXSPI` or `CONFIG_CODE_FLEXSPI2` is selected.

File: boards/arm/mimxrt1020_evk/Kconfig.defconfig [hunk 2]
```diff
@@ -15,12 +15,16 @@ endchoice
...
+
 config FLASH_MCUX_FLEXSPI_NOR
-	default y if FLASH && BOARD_MIMXRT1020_EVK
+	default y
 
 config FLASH_MCUX_FLEXSPI_XIP
-	default y if FLASH && BOARD_MIMXRT1020_EVK
-	depends on MEMC_MCUX_FLEXSPI
+	default y
+	depends on (MEMC_MCUX_FLEXSPI && (CODE_FLEXSPI || CODE_FLEXSPI2))
```
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply boards/arm/mimxrt1020_evk/Kconfig.defconfig]: I've updated the PR to fix the KConfig definition at the driver level, and remove it at the board level

**Review by mmahadevan108 — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**